manifests/init.pp

```diff
@@ -93,6 +182,22 @@
   Boolean                                   $restart_on_config_change  = true,
   Boolean                                   $restart_on_package_change = true,
 ) {
+  $default_notifications_notifications = lookup('opensearch::default_notifications_notifications', Hash)
+  $default_notifications_notifications_core = lookup('opensearch::default_notifications_notifications_core', Hash)
+  $default_observability_observability = lookup('opensearch::default_observability_observability', Hash)
+  $default_reporting_reports_scheduler = lookup('opensearch::default_reporting_reports_scheduler', Hash)
+
+  $default_security_action_groups = lookup('opensearch::default_security_action_groups', Hash)
+  $default_security_allowlist = lookup('opensearch::default_security_allowlist', Hash)
+  $default_security_audit = lookup('opensearch::default_security_audit', Hash)
+  $default_security_config = lookup('opensearch::default_security_config', Hash)
+  $default_security_internal_users = lookup('opensearch::default_security_internal_users', Hash)
+  $default_security_nodes_dn = lookup('opensearch::default_security_nodes_dn', Hash)
+  $default_security_roles_mapping = lookup('opensearch::default_security_roles_mapping', Hash)
+  $default_security_roles = lookup('opensearch::default_security_roles', Hash)
+  $default_security_tenants = lookup('opensearch::default_security_tenants', Hash)
+  $default_security_whitelist = lookup('opensearch::default_security_whitelist', Hash)
+
```

Comment:
These are not added as parameters and are explicitly `lookup`ed for in order to avoid the users updating them.

We should probably do the same for the `default_settings` parameter but this is a backwards incompatible change and maybe there would be concerns about it?

Reply:
In any case the default_settings change would deserve it's own backwards-incompatible PR for a better changelog, and can be done anytime, not necessarily with the next release, so I am not concerned much about this.